### PR TITLE
Use SoA for `SimTrackView` state data

### PR DESCRIPTION
**Pull request by esseivaju**
`SimTrackView` access data using a SoA. `SimStateData` is now a SoA instead of having a `state` AoS attribute. Renamed `SimTrackState` to `SimTrackInitializer` as it's still useful to pack arguments of `SimTrackView::operator=(SimTrackInitializer const&)` and minimize changes required in `InitTracksLauncher` and `ProcessPrimariesLauncher`

### CODE REVIEW

File: src/celeritas/track/SimData.hh [hunk 1]
```diff
@@ -151,11 +171,37 @@ template<MemSpace M>
...
+        celeritas::StateCollection<value_type, Ownership::value, MemSpace::host>
+            state;
+        std::vector<value_type> initial_state(size, init);
+        make_builder(&state).insert_back(initial_state.begin(),
+                                         initial_state.end());
+        state_collection = state;
+    };
+    resize_state_collection(data->track_ids);
+    resize_state_collection(data->parent_ids);
+    resize_state_collection(data->event_ids);
```
Comment (amandalund) [inline src/celeritas/track/SimData.hh]: We should just be able to use the `resize()` helper function for these:
```suggestion
    resize(&data->event_ids, size);
```

File: src/celeritas/track/SimData.hh [hunk 2]
```diff
@@ -151,11 +171,37 @@ template<MemSpace M>
...
+                                                       auto init) {
+        using value_type = typename std::remove_reference<
+            decltype(state_collection)>::type::value_type;
+        celeritas::StateCollection<value_type, Ownership::value, MemSpace::host>
+            state;
+        std::vector<value_type> initial_state(size, init);
+        make_builder(&state).insert_back(initial_state.begin(),
+                                         initial_state.end());
+        state_collection = state;
+    };
```
Comment (amandalund) [inline src/celeritas/track/SimData.hh]: We might be able to use a `resize()` and `fill()` here instead of constructing on the host and copying?

**Review by amandalund — COMMENTED** [review]
Nice @esseivaju!

File: src/celeritas/track/SimData.hh (hunk 2 above)
Comment (esseivaju) [reply src/celeritas/track/SimData.hh]: Good point, thanks for the feedback!

File: src/corecel/data/detail/Filler.cu [hunk 3]
```diff
@@ -5,6 +5,8 @@
 //---------------------------------------------------------------------------//
 //! \file corecel/data/detail/Filler.cu
 //---------------------------------------------------------------------------//
+#include "celeritas/Types.hh"
```
Comment (sethrj) [inline src/corecel/data/detail/Filler.cu]: `corecel` is upstream of `celeritas` so we unfortunately can't include it here. You should be able to make a `celeritas/track/detail/Filler.cu` that is effectively like this file but only includes instantiation for the `celeritas/Types` that you need.

**Review by sethrj — COMMENTED** [review]
Looks great with the exception of the cyclic dependency in Filler.

**Comment by sethrj** [comment]
Oh goodness, HIP is failing again... :(

**Review by sethrj — APPROVED** [review]

**Comment by esseivaju** [comment]
Looks like one test is failing with HIP but working fine otherwise. Not sure I'll have much time to investigate that this week..

**Review by amandalund — APPROVED** [review]
Thanks @esseivaju! Don't worry about the HIP test, it's been failing sporadically (we're not sure why but it's not related to your changes).

**Comment by esseivaju** [comment]
Looks like HIP just doesn't want to cooperate.

**Pull request merged**